jcl/src/java.base/share/classes/jdk/internal/vm/Continuation.java

```diff
@@ -24,13 +24,21 @@
 
 import java.util.Set;
 import java.util.function.Supplier;
+import jdk.internal.access.JavaLangAccess;
+import jdk.internal.access.SharedSecrets;
 
 /**
  * Continuation class performing the mount/unmount operation for VirtualThread
  */
 public class Continuation {
 	private long vmRef; /* J9VMContinuation */
 
+	private final ContinuationScope scope;
+	private final Runnable runnable;
+	private Continuation parent;
+	private boolean started;
+	private boolean finished;
```

Comment:
one more thing, this needs to be volatile or it needs to be set outside the vm (like in the finally block)